### ISSUE TITLE
loss function design

### ISSUE THREAD

**Issue by HuaizeLiu** [tvaranka/fineface#3]
Model performance is great. Will the training code be released? How was the loss function designed during the training process?

**Comment by tvaranka** [comment]
There is a plan for releasing the training code with the generated data, but it will take some time to clean up everything.

The loss function is just the standard L2 diffusion loss, no extra terms are used.

**Comment by ruotianwuf** [comment]
> 有一个计划来发布包含生成数据的训练代码，但需要一些时间来清理所有内容。
> 
> 损失函数只是标准的 L2 扩散损失，没有使用额外的术语。

Hello,

Can you provide an exact time for the release of the train code and dataset? I really need this, as it will significantly reduce my current workload.

Best wishes!

**Comment by tvaranka** [comment]
> > 有一个计划来发布包含生成数据的训练代码，但需要一些时间来清理所有内容。
> > 损失函数只是标准的 L2 扩散损失，没有使用额外的术语。
> 
> Hello,
> 
> Can you provide an exact time for the release of the train code and dataset? I really need this, as it will significantly reduce my current workload.
> 
> Best wishes!

Unfortunately I am unable to provide an exact date as of this moment. I will do my best to have it ready in the near future.

Thanks

**Comment by wzh125** [comment]
Hi，When will you provide the train code and dataset?

**Comment by tvaranka** [comment]
> Hi，When will you provide the train code and dataset?

The dataset is released, please see [code](https://github.com/tvaranka/fineface/blob/9c104c3fd6327e18f42b92a8997493f38b0ba2e1/fineface/au_dataset.py) for data preprocessing and [metadata](https://huggingface.co/Tvaranka/fineface/tree/main) of the dataset. The image/video data for affectnet/DISFA can be downloaded from their original sources. The training code will be released very soon.

Thanks!

**Comment by tvaranka** [comment]
Training script added
